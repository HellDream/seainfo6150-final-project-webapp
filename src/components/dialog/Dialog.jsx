import React from 'react';
import styles from './Dialog.module.css';

const Dialog = (props) => {
    const message = props.message;
    const open = props.open;
    const onOk = props.ok;
    let dialog = (
        <div className={styles.container}>
            <div
                role="dialog"
                aria-modal={true}
                aria-labelledby="dialog_label"
                className={styles.dialogWrapper}
            >
                <button
                    className={styles.closeBtn}
                    onClick={() => props.onClose(!open)}
                    autoFocus={true}
                ></button>
                <h3 className={styles.content}>{message}</h3>
                {onOk && (
                    <button className={styles.okBtn} onClick={onOk}>
                        OK
                    </button>
                )}
            </div>
        </div>
    );
    if (!open) {
        dialog = null;
    }
    return open ? dialog : <div></div>;
};

export default Dialog;
