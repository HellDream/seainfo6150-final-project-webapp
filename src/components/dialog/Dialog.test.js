import React from 'react';
import Dialog from './Dialog';
import { render } from '@testing-library/react'

describe('CategoryItem tests', () => {
    it('renders correctly', () => {
        const { container } = render(<Dialog message="message" open={true} onClose={(open)=>{}} ok={()=>{}} />);
        expect(container).toMatchSnapshot();
    });
});
  