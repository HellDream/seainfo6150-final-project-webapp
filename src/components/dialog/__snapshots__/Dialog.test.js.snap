// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryItem tests renders correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      aria-labelledby="dialog_label"
      aria-modal="true"
      class="dialogWrapper"
      role="dialog"
    >
      <button
        class="closeBtn"
      />
      <h3
        class="content"
      >
        message
      </h3>
      <button
        class="okBtn"
      >
        OK
      </button>
    </div>
  </div>
</div>
`;
